### PR TITLE
FHAC-1232 modified files replaced system.out with LOG.debug and excep…

### PR DESCRIPTION
**Pull request by sjarral112**
FHAC-1232 modified files replaced system.out with LOG.debug and exceptions to LOG.error

Files modified 

        modified:   Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java
        modified:   Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/devtools/FindAndReplaceFile.java
        modified:   Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/patientcorrelation/nhinc/parsers/helpers/CreationTimeHelper.java
        modified:   Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/policy/XDRPolicyTransformHelper.java
        modified:   Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java
        modified:   Product/Production/Services/DocumentSubmissionCore/

Reviewers : @mhpnguyen @TabassumJafri

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java [hunk 1]
```diff
@@ -69,14 +69,14 @@ public boolean handleMessage(SOAPMessageContext messageContext) {
...
-                        System.out.println("Detected an asynchronous request message");
+                        LOG.debug("Detected an asynchronous request message");
                         // Override the Message Id field
                         String messageId = null;
 
                         if (messageContext.containsKey(NhincConstants.ASYNC_MESSAGE_ID_PROP) == true) {
                             messageId = (String) messageContext.get(NhincConstants.ASYNC_MESSAGE_ID_PROP);
 
-                            System.out.println("Setting message ID to " + messageId);
+                            LOG.debug("Setting message ID to " + messageId);
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java]: use this format LOG.debug("Setting message ID to {}" , messageId);

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java [hunk 2]
```diff
@@ -85,24 +85,24 @@ public boolean handleMessage(SOAPMessageContext messageContext) {
...
-                        System.out.println("Detected an asynchronous response message");
+                        LOG.debug("Detected an asynchronous response message");
                         // Override the Relates To Id field
                         String relatesToId = null;
 
                         if (messageContext.containsKey(NhincConstants.ASYNC_RELATES_TO_PROP) == true) {
                             relatesToId = (String) messageContext.get(NhincConstants.ASYNC_RELATES_TO_PROP);
 
-                            System.out.println("Setting relates to ID to " + relatesToId);
+                            LOG.debug("Setting relates to ID to " + relatesToId);
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java]: update the logging format.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/devtools/FindAndReplaceFile.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ public static int searchAndReplace(File fDirToLook, File fFileName) throws Throw
                 try {
                     boolean bCopied = copy(fFileName, fDirToLook);
                     if (bCopied) {
-                        System.out.println("Replaced file: " + fDirToLook.getCanonicalPath());
+                       LOG.debug("Replaced file: " + fDirToLook.getCanonicalPath());
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/devtools/FindAndReplaceFile.java]: update logging format.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/policy/XDRPolicyTransformHelper.java [hunk 4]
```diff
@@ -133,16 +133,16 @@ private String getIdentifiersFromRequest(ProvideAndRegisterDocumentSetRequestTyp
...
 
         for (int x = 0; x < object.getIdentifiable().size(); x++) {
-            System.out.println(object.getIdentifiable().get(x).getName());
+            LOG.debug(object.getIdentifiable().get(x).getName());
 
             if (object.getIdentifiable().get(x).getDeclaredType().equals(RegistryPackageType.class)) {
                 RegistryPackageType registryPackage = (RegistryPackageType) object.getIdentifiable().get(x).getValue();
 
-                System.out.println(registryPackage.getSlot().size());
+                LOG.debug(registryPackage.getSlot().size());
```
Comment (TabassumJafri) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/policy/XDRPolicyTransformHelper.java]: Please add some message like "Slot(s) in registry Package is"

**Review by TabassumJafri — CHANGES_REQUESTED** [review]
Please see my inline comments.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java [hunk 5]
```diff
@@ -281,7 +281,7 @@ public static PersonNameType translatePNListtoPersonNameType(List<PNExplicit> na
                     } else {
                         nameString = strValue;
                     }
-                    System.out.println("Here is the contents of the string: " + strValue);
+                    LOG.debug("Here is the contents of the string: " + strValue);
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java]: Consider to use LOG.debug("custom msg {}",strValue).

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java [hunk 6]
```diff
@@ -352,7 +352,7 @@ public static PersonNameType translateENListtoPersonNameType(List<ENExplicit> na
...
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java]: please see previous comments

**Review by mhpnguyen — CHANGES_REQUESTED** [review]

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java [hunk 7]
```diff
@@ -69,14 +69,14 @@ public boolean handleMessage(SOAPMessageContext messageContext) {
...
-                        System.out.println("Detected an asynchronous request message");
+                        LOG.debug("Detected an asynchronous request message");
                         // Override the Message Id field
                         String messageId = null;
 
                         if (messageContext.containsKey(NhincConstants.ASYNC_MESSAGE_ID_PROP) == true) {
                             messageId = (String) messageContext.get(NhincConstants.ASYNC_MESSAGE_ID_PROP);
 
-                            System.out.println("Setting message ID to " + messageId);
+                            LOG.debug("Setting message ID to {}" + messageId);
```
Comment (christophermay07) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java]: Here and below: Do not mix parameterized logging with String concatenation.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java [hunk 8]
```diff
@@ -85,24 +85,24 @@ public boolean handleMessage(SOAPMessageContext messageContext) {
...
-                        System.out.println("Detected an asynchronous response message");
+                        LOG.debug("Detected an asynchronous response message");
                         // Override the Relates To Id field
                         String relatesToId = null;
 
                         if (messageContext.containsKey(NhincConstants.ASYNC_RELATES_TO_PROP) == true) {
                             relatesToId = (String) messageContext.get(NhincConstants.ASYNC_RELATES_TO_PROP);
 
-                            System.out.println("Setting relates to ID to " + relatesToId);
+                            LOG.debug("Setting relates to ID to {}" + relatesToId);
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageHandler.java]: same thing here.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/devtools/FindAndReplaceFile.java [hunk 9]
```diff
@@ -128,7 +128,7 @@ public static int searchAndReplace(File fDirToLook, File fFileName) throws Throw
                 try {
                     boolean bCopied = copy(fFileName, fDirToLook);
                     if (bCopied) {
-                        System.out.println("Replaced file: " + fDirToLook.getCanonicalPath());
+                       LOG.debug("Replaced file: {}" + fDirToLook.getCanonicalPath());
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/devtools/FindAndReplaceFile.java]: same thing here. consider to use parameterized logging.

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java [hunk 10]
```diff
@@ -281,7 +281,7 @@ public static PersonNameType translatePNListtoPersonNameType(List<PNExplicit> na
                     } else {
                         nameString = strValue;
                     }
-                    System.out.println("Here is the contents of the string: " + strValue);
+                    LOG.debug("Here is the contents of the string: {}" + strValue);
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java]: consider to use parameterize logging

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java [hunk 11]
```diff
@@ -352,7 +352,7 @@ public static PersonNameType translateENListtoPersonNameType(List<ENExplicit> na
...
```
Comment (mhpnguyen) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/subdisc/HL7Extractors.java]: consider to use parameterized logging

**Review by mhpnguyen — CHANGES_REQUESTED** [review]

**Review by mhpnguyen — APPROVED** [review]

**Review by TabassumJafri — APPROVED** [review]
Approve without any comments

**Pull request merged**